scripts/sync-algolia-index.ts

```diff
@@ -27,33 +33,26 @@ const getFileInfos = (
     type: BlockProtocolFileTypes;
   }>,
   type: BlockProtocolFileTypes,
-): {
-  inputPath: string;
-  outputPath: string;
-  type: BlockProtocolFileTypes;
-}[] => {
+): FileTypeAndPaths[] => {
   const files = fs.readdirSync(dirPath);
 
-  let newArrayOfFiles = arrayOfFiles || [];
+  const newArrayOfFiles: FileTypeAndPaths[] = [];
 
   files.forEach((file) => {
     if (fs.statSync(`${dirPath}/${file}`).isDirectory()) {
-      newArrayOfFiles = getFileInfos(
-        `${dirPath}/${file}`,
-        newArrayOfFiles,
-        type,
-      );
-    } else {
-      const inputPath = path.join(dirPath, "/", file);
-      const outputPath = `.\\output\\${path.join(
-        "./output/",
-        dirPath,
-        "/",
-        file,
-      )}`;
-      if (inputPath.endsWith(".md") || inputPath.endsWith(".mdx")) {
-        newArrayOfFiles.push({ inputPath, outputPath, type });
-      }
+      // the only subfolder, 3_spec, is indexed as a different type
+      return;
+    }
+
+    const inputPath = path.join(dirPath, "/", file);
+    const outputPath = `.\\output\\${path.join(
+      "./output/",
+      dirPath,
+      "/",
+      file,
+    )}`;
```

Comment:
Hmmm not sure I understand why we have `path.join` inside ``` `` ``` 🤔

Reply:
I expect we will heavily rewrite this soon as I'd rather not start improving bits like this